internal/services/pubkey_service.go

```diff
@@ -99,6 +117,48 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	resources, err := pubkeyResource.ListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			arn, err := sourcesClient.GetArn(r.Context(), res.SourceID)
+			if err != nil {
+				if errors.Is(err, sources.ApplicationNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: application not found", err, 404))
+					return
+				}
+				if errors.Is(err, sources.AuthenticationForSourcesNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: authentication not found", err, 404))
+					return
+				}
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources", err, 500))
+				return
+			}
+			crd, err := stsClient.AssumeRole(arn)
+			if err != nil {
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "cannot assume role: %w", err, 500))
+				return
+			}
```

Comment:
It looks like we have a common pattern across our codebase, there are three other places where we create client, get arn from sources and assume role. I believe we should refactor this into a single function, not here, not in this PR, but this would be a nice PR for you to work on.

The question is where to put it, since this operation uses both AWS client and sources client. I guess putting it into AWS is reasonable, this will create a dependency (AWS needs Sources package) but I think it is reasonable. @ezr-ondrej ?